rs/consensus/idkg/src/pre_signer.rs

```diff
@@ -390,143 +390,147 @@ impl IDkgPreSignerImpl {
             target_subnet_xnet_transcripts.insert(transcript_params_ref.transcript_id);
         }
 
-        let mut valid_dealing_supports = self.validated_dealing_supports.write().unwrap();
-        let mut ret = Vec::new();
-        for (id, support) in idkg_pool.unvalidated().dealing_support() {
-            // Dedup dealing support by checking whether a previous (transcript_id, dealer_id,
-            // dealing_hash) was already signed by the signer
-            let key = IDkgValidatedDealingSupportIdentifier::from(&support);
-            let maybe_signers = valid_dealing_supports.get_mut(&key);
-            if maybe_signers
-                .as_ref()
-                .is_some_and(|signers| signers.contains(&support.sig_share.signer))
-            {
-                ret.push(IDkgChangeAction::HandleInvalid(
-                    id,
-                    format!("Duplicate dealing support in unvalidated batch: {support}"),
-                ));
-                continue;
-            };
-            // Drop shares for xnet reshare transcripts
-            if source_subnet_xnet_transcripts.contains(&support.transcript_id) {
-                self.metrics.pre_sign_errors_inc("xnet_reshare_support");
-                ret.push(IDkgChangeAction::HandleInvalid(
-                    id,
-                    format!("Support for xnet reshare transcript: {support}"),
-                ));
-                continue;
-            }
+        let unvalidated_supports: Vec<_> = idkg_pool.unvalidated().dealing_support().collect();
```

Comment:
I assume it is not possible to call `into_par_iter` directly on `dealing_support`?

Maybe we could mention that here